code/modules/jobs/access.dm

```diff
@@ -45,11 +45,12 @@
 		var/mob/living/simple_animal/animal = accessor
 		if(check_access(animal.access_card))
 			return TRUE
-	else if(isbrain(accessor) && istype(accessor.loc, /obj/item/mmi))
-		var/obj/item/mmi/brain_mmi = accessor.loc
-		if(ismecha(brain_mmi.loc))
-			var/obj/vehicle/sealed/mecha/big_stompy_robot = brain_mmi.loc
-			return check_access_list(big_stompy_robot.accesses)
+	else if(isbrain(accessor))
+		var/obj/item/mmi/brain_mmi = get(accessor.loc, /obj/item/mmi)
+		if(brain_mmi)
+			if(ismecha(brain_mmi.loc))
+				var/obj/vehicle/sealed/mecha/big_stompy_robot = brain_mmi.loc
+				return check_access_list(big_stompy_robot.accesses)
```

Comment:
```suggestion
	else if(isbrain(accessor))
		var/obj/item/mmi/brain_mmi = get(accessor.loc, /obj/item/mmi)
		if(brain_mmi && ismecha(brain_mmi.loc))
			var/obj/vehicle/sealed/mecha/big_stompy_robot = brain_mmi.loc
			return check_access_list(big_stompy_robot.accesses)
```

this saves us some ugly indentation.

also `get()` is interesting, do we use that in other spots to grab the MMI from a location? i ask because something like `locate()` or of that type is typically used. this is probably fine htough

Reply:
get works on parent location recursifly, locate can be from anywhere in the target, so it feels safer to use it this way. humans get filterd out before tho so I dont see it going wrong any time soon